src/aurelia-slickgrid/services/universalTranslate.service.ts

```diff
@@ -0,0 +1,39 @@
+import { TranslaterService, TranslateServiceEventName } from '@slickgrid-universal/common';
+import { inject } from 'aurelia-framework';
+import { I18N } from 'aurelia-i18n';
+
+interface Locales {
+  [locale: string]: string;
+}
+
+interface TranslateOptions {
+  loadPath: string;
+  lang: string;
+}
+
+/**
+ * This is a Translate Service Wrapper for Slickgrid-Universal,
+ * it must implement Slickgrid-Universal TranslaterService interface to work properly
+ */
+@inject(I18N)
+export class UniversalTranslateService implements TranslaterService {
+  eventName = 'onLanguageChange' as TranslateServiceEventName;
+
+  constructor(private i18n: I18N) { }
+
+  getCurrentLanguage(): string {
+    return this.i18n.getLocale();
+  }
+
+  async use(newLang: string): Promise<Locales> {
+    return this.i18n.setLocale(newLang);
```

Comment:
what was the idea with this @ghiscoding? setLocale will return the translation function. I'm not sure I get the intention of this code part. What are those `Locale` supposed to be?

Reply:
The `Locale` is when the user wants to ignore I18N and define his own Custom Locale, so basically it's when they use a Single Locale, then they don't need to use I18N. English Locale is defined as the default, but you could create a custom one like German Locale and load that Locale (`german.ts`) and they're done. Here's the [Wiki](https://github.com/ghiscoding/aurelia-slickgrid/wiki/Localization-with-Custom-Locales), I know the name might be confusing, I just didn't know what other name to use for representing a locale loading without using I18N.

Reply:
Ok that means you'd like to return the loaded json translation there? Like an object with keynames being locales and values the loaded resources?

Reply:
oh wait sorry I didn't notice you commented in the Translate wrapper file, I thought you were in another file. I think you are right, this is wrong and shouldn't return a `Locale` (since that is what I described in previous comment and doesn't fit with I18N). If you have the correct Type from I18N please change it

 Here is how I use it in my other lib
![image](https://user-images.githubusercontent.com/643976/101170321-48927f80-360c-11eb-8709-af00eded65f2.png)

and yes you can see I simply load a json file with `require(x.json)`, nothing fancy and I know it might just work with WebPack but it does work with it.

Reply:
Sure I'm still not sure what the returned value of use should be since the type Locales isnt really self-explanatory. As asked above an object keynames as Locales and their value the loaded json or the path to those? Perhaps you can give me an example of how the returned object should ideally look like and I can find you the proper i18next API for it

Reply:
It can, and probably should, be the same as what you declared in Au-I18N, which seems to be ` setLocale(locale: string): Promise<i18next.TFunction>` or just put it as `Promise<any>` (what you had in previous Au-I18N). I rarely care or use the return variable anyway, but I do care that it's returning a Promise since I very often want to `await` it (for spinner or text in UI).

For example in 1 of the Aurelia-Slickgrid [demo 12](https://ghiscoding.github.io/aurelia-slickgrid/#/slickgrid/example12), I just await the `setLocale` to display the new locale in the UI next to the switch language button (see below).
![image](https://user-images.githubusercontent.com/643976/101193900-3a079080-362b-11eb-89d8-95a0bb803e23.png)
![image](https://user-images.githubusercontent.com/643976/101194127-9074cf00-362b-11eb-9a16-69f3cd638bca.png)

and inside my Slickgrid-Universal interface, I made it return `Promise<any> | any` (I think the `any` was for Angular and I expect Aurelia to return a Promise). However I'm not sure that the description I wrote for the return type is valid, it's kinda of a mix of the Au-I18N and Angular ngx-translate.

So to answer your question, I'm indifferent on the return, so long as it's a Promise with possibly something inside, making it return the same as Au-I18N would be perfect. 

![image](https://user-images.githubusercontent.com/643976/101194415-f9f4dd80-362b-11eb-9834-70cd6f2d9f51.png)

Reply:
No question about the Promise I guessed you'd need to await it so clear on that. What does your angular version with ngx-translate return for the same function? Casting any and returning the Translation function from aurelia-i18n is definitely the simplest solution but kinda beats the purpose of a unified API. Since we dont care I'd really like it to do similar to others e.g angular

Reply:
So if I remember correctly it returns the translation set of the associated locale (which is pretty much what I wrote in my `use()` interface) and just to be sure I looked at ngx-translate code and that is exactly it as shown on this [line](https://github.com/ngx-translate/core/blob/master/projects/ngx-translate/core/src/lib/translate.service.ts#L203). Now you see where I copied the `use()` method name from, mostly because I wanted the shortest name :P 

Here's a small print screen of this [line](https://github.com/ngx-translate/core/blob/master/projects/ngx-translate/core/src/lib/translate.service.ts#L203) of code, there's a but more code because I think you can override it, but for the most part that is it.

![image](https://user-images.githubusercontent.com/643976/101213941-2d923080-3649-11eb-80d0-4209881bb882.png)

Reply:
Hey so I made a few commits and wanted the build to pass so I went ahead and made few changes in this file. What I found out was that the `setup(options)` is completely unnecessary, it existed because I copied this file over from Slickgrid-Universal which is a full implementation (or rather very basic) of a Translate Service like I18N is, so that method must exist in the monorepo only for demo purposes and is not needed at all here, so I removed it. I also went ahead and fixed a few of the other comments you've put, if you can take another quick look at it that'd be great (I'm done with the file, so you change whatever else in that file if you wish). Thanks again for the review, your feedback helps **a lot**, really appreciate

Reply:
Sure no prob. Alright thx for the pointers, I'm not sure but perhaps I can get to it today midday when my little boy sleeps.

Reply:
I always wonder what TZ you actually are :P ...I'm in Montreal Canada (East TZ), I think we have time diff but I always wonder how many :P

Reply:
Hehe yeah I'm from Austria, GMT+1 so I think 6-7 hours diff. Jeez you talking about Montreal makes me envy. Definitely want to visit that great place again.

Reply:
Ok looking at this in detail now I think your approach is right. I'd keep it as promise<any> and the return type as is, since there is not really lots of benefit returning the same nextLang that you gave the function.